app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java

```diff
@@ -123,17 +126,24 @@ public void onCreate(Bundle savedInstanceState) {
 
             super.onCreate(savedInstanceState);
             try {
-                getWindow()
-                        .getDecorView()
-                        // TODO: Deprecated setSystemUiVisibility(int) and related flags,
-                        //  see https://github.com/Igalia/wolvic/issues/800
-                        .setSystemUiVisibility(
-                                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-                                        | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                                        | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                                        | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-                                        | View.SYSTEM_UI_FLAG_FULLSCREEN
-                                        | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    getWindow().setDecorFitsSystemWindows(false);
+                    WindowInsetsController controller = getWindow().getInsetsController();
+                    if(controller != null) {
+                        controller.hide(WindowInsets.Type.statusBars() | WindowInsets.Type.navigationBars());
+                        controller.setSystemBarsBehavior(WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
+                    }
+                } else {
+                    getWindow()
+                            .getDecorView()
+                            .setSystemUiVisibility(
+                                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                                            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                                            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                                            | View.SYSTEM_UI_FLAG_FULLSCREEN
+                                            | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+                }
```

Comment:
Correct me if I'm missing something but can't we just simply call `SetImmersiveSticky()`  here and avoid this code duplication?

Reply:
I'm not sure how we can re-use the `SetImmersiveSticky()`. By the way, the code duplication exists in the original code and is not introduced by this PR.

Reply:
Forget about that, I didn't realize that the other was in the offscreen display code, I somehow assumed that everything was inside the PlatformActivity